### PR TITLE
Bugfix: Removed @RequirePost and changed ssh key migration behavior

### PR DESCRIPTION
**Pull request by djesionek**
I changed the behavior from my feature added in #445 as suggested by @daniel-beck by moving the migration code to EC2Cloud.readResolve(). I also tested again the migration from 1.51 to 1.52. I could confirm that it wasn't working before and that it works now for me with that change.
Not sure why the @RequirePost lead to issues as it was requested as an addition but maybe someone can clarify if there is a better way than just removing it for now.

Thanks to @varyvol for the bugnotice and @daniel-beck for the hint how to fix it quickly.

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/EC2Cloud.java [hunk 1]
```diff
@@ -1066,7 +1065,6 @@ public ListBoxModel doFillSshKeysCredentialsIdItems(@QueryParameter String sshKe
                     .includeCurrentValue(sshKeysCredentialsId);
         }
 
-        @RequirePOST
```
Comment (MRamonLeon) [inline src/main/java/hudson/plugins/ec2/EC2Cloud.java:1069]: It looks like fine to remove this annotation here, the method is just returning whether the credential-id showed is an actual rsa private key. But better to confirm with some security person: @daniel-beck, @Wadeck, ...

Comment (daniel-beck) [reply src/main/java/hudson/plugins/ec2/EC2Cloud.java:1069]: There's still a permission check.

The docs for this are at https://www.jenkins.io/doc/developer/security/form-validation/ -- please read and if anything specific is unclear, ask.

Comment (Wadeck) [reply src/main/java/hudson/plugins/ec2/EC2Cloud.java:1069]: getSshCredential retrieves credentials, but as the RequirePOST protects against CSRF, there is no issue. The method is already checking for ADMINISTER.

Side note, such methods should not have troubles to use POST for the check. You can change the https://github.com/jenkinsci/ec2-plugin/blob/8702cc4a89bd7c096bba07211e222cde6d931f5d/src/main/resources/hudson/plugins/ec2/AmazonEC2Cloud/config-entries.jelly#L39 (and similar for Eucalyptus) by adding `checkMethod="post"`

**Comment by Wadeck** [comment]
@markyjackson-taulia Please read the comment above about the RequirePOST ;)
> A @RequirePOST is required when doing doCheckX
(from https://github.com/jenkinsci/ec2-plugin/pull/445#pullrequestreview-416564695)

----

> Not sure why the @RequirePost lead to issues as it was requested as an addition

Be sure to test your code before commiting, there is no "manual test" performed everytime on PR and this could lead to broken release sometimes ;)

**Comment by djesionek** [comment]
Now I tested the migration manually after all changes maybe we should release that as a hotfix or something before more updates fail.

**Comment by varyvol** [comment]
@MRamonLeon @res0nance @daniel-beck Tests are passing and even if no test was created to verify the issue, @djesionek has tested it manually. Should we maybe release this and work in the test later?

**Review by res0nance — APPROVED** [review]

**Comment by res0nance** [comment]
I think releasing as a bugfix release is reasonable. waiting to see if there are any other comments

**Comment by batmat** [comment]
I think that if really we've had people manually verifying the fix, we _may_ want to merge.

However, given the relative ease of use of `@LocalData`, I would recommend we also have a test accompanying this fix so it's caught next time.
* Docs: https://javadoc.jenkins.io/component/jenkins-test-harness/org/jvnet/hudson/test/recipes/LocalData.html
* Working example: https://github.com/batmat/localdata-jenkinsrule/

**Comment by varyvol** [comment]
See #500 for an automated test.

**Review by MRamonLeon — APPROVED** [review]

**Pull request merged**